API.md

```diff
@@ -0,0 +1,552 @@
+
+await libp2p._dht.put(key, value)
+```
+
+### _dht.get
+
+Queries the DHT for a value stored for a given key.
+
+`libp2p._dht.get(key, options)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| key | `String` | key to get from the dht |
+| [options] | `Object` | get options |
+| [options.timeout] | `number` | maximum time the query should run |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<Buffer>` | Value obtained from the DHT |
+
+#### Example
+
+```js
+// ...
+
+const key = '/key'
+const value = await libp2p._dht.get(key)
+```
+
+### _dht.getMany
+
+Queries the DHT for the n values stored for the given key (without sorting).
+
+`libp2p._dht.getMany(key, nvals, options)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| key | `String` | key to get from the dht |
+| nvals | `number` | number of values aimed |
+| [options] | `Object` | get options |
+| [options.timeout] | `number` | maximum time the query should run |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<Array<{from: PeerId, val: Buffer}>>` | Array of records obtained from the DHT |
+
+#### Example
+
+```js
+// ...
+
+const key = '/key'
+const { from, val } = await libp2p._dht.get(key)
+```
+
+### pubsub.getPeersSubscribed
+
+Gets a list of the peer-ids that are subscribed to one topic.
+
+`libp2p.pubsub.getPeersSubscribed(topic)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| topic | `string` | topic to publish |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Array<String>` | peer-id subscribed to the topic |
+
+#### Example
+
+```js
+const peerIds = libp2p.pubsub.getPeersSubscribed(topic)
+```
+
+### pubsub.getTopics
+
+Gets a list of topics the node is subscribed to.
+
+`libp2p.pubsub.getTopics()`
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Array<String>` | topics the node is subscribed to |
+
+#### Example
+
+```js
+const topics = libp2p.pubsub.getTopics()
+```
+
+### pubsub.publish
+
+Publishes messages to the given topics.
+
+`libp2p.pubsub.publish(topic, data)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| topic | `string` | topic to publish |
+| data | `Buffer` | data to publish  |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise` | publish success |
+
+#### Example
+
+```js
+const topic = 'topic'
+const data = Buffer.from('data')
+
+await libp2p.pubsub.publish(topic, data)
+```
+
+### pubsub.subscribe
+
+Subscribes the given handler to a pubsub topic.
+
+`libp2p.pubsub.subscribe(topic)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| topic | `string` | topic to subscribe |
+| handler | `function(<Object>)` | handler for new data on topic |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `void` |  |
```

Comment:
Just checking this is not an async operation?

Reply:
no, we do not need any async operation for any of the pubsub routers in this operation